modules/basic-skills/src/views/full/slot.jsx

```diff
@@ -0,0 +1,277 @@
+    })
+  }
+
+  fetchActions = () => {
+    this.props.bp.axios.get(`/actions`).then(({ data }) => {
+      this.setState({
+        actions: data.filter(action => !action.metadata.hidden)
+      })
+    })
+  }
+
+  getSelectedIntent() {
+    const intentName = this.state.selectedIntentOption && this.state.selectedIntentOption.value
+    return intentName && this.state.intents.find(x => x.name === intentName)
+  }
+
+  getSelectedSlot(intent) {
+    const slotName = this.state.selectedSlotOption && this.state.selectedSlotOption.value
+    return intent && intent.slots.find(x => x.name === slotName)
+  }
+
+  isFormValid() {
+    return (
+      this.state.selectedIntentOption &&
+      this.state.selectedSlotOption &&
+      this.state.contentElement &&
+      this.state.notFoundElement
+    )
+  }
+
+  validateIntentExists = intentName => {
+    if (!intentName) {
+      return
+    }
+
+    const exists = this.state.intents.find(x => x.name === intentName)
+    if (!exists) {
+      this.setState({ error: 'Missing intent: This intent does not exists anymore!' })
+    }
+  }
+
+  validateSlotExists = (intentName, slotName) => {
+    if (!intentName || !slotName) {
+      return
+    }
+
+    const currentIntent = this.state.intents.find(x => x.name === intentName)
+    if (!currentIntent || !currentIntent.slots.find(x => x.name === slotName)) {
+      this.setState({ error: 'Missing slot: This slot does not exits anymore!' })
+    }
+  }
+
+  handleContentChange = item => {
+    this.setState({ contentElement: item.id })
+  }
+
+  handleSlotChange = selectedSlotOption => {
+    this.validateSlotExists(this.state.selectedIntentOption.value, selectedSlotOption.value)
+    this.setState({ selectedSlotOption })
+  }
+
+  handleIntentChange = selectedIntentOption => {
+    this.validateIntentExists(selectedIntentOption.value)
+    this.setState({ selectedIntentOption, selectedSlotOption: undefined })
+  }
+
+  handleNotFoundChange = item => {
+    this.setState({ notFoundElement: item.id })
+  }
+
+  handleMaxRetryAttemptsChange = event => {
+    const value = Number(event.target.value)
+    if (value > MAX_RETRIES) {
+      this.setState({ error: `Too many retry attempts: Choose a number less than or equal to ${MAX_RETRIES}` })
+    } else {
+      this.setState({ maxRetryAttempts: value })
+    }
+  }
+
+  handleActionChange = value => {
+    this.setState({ validationAction: { value, label: value } })
+  }
+
+  toggleAddValidation = () => {
+    this.setState({ addValidation: !this.state.addValidation })
+  }
+
+  getSlotOptionsForIntent(intent) {
+    return (
+      intent &&
+      intent.slots.map(slot => {
+        return { value: slot.name, label: slot.name }
+      })
+    )
+  }
+
+  render() {
+    const intent = this.getSelectedIntent()
+    const intentsOptions = this.state.intents.map(intent => {
+      return { value: intent.name, label: intent.name }
+    })
+    const slotOptions = this.getSlotOptionsForIntent(intent)
```

Comment:
Just wondering if those maps couldn't be set elsewhere (like when their values changes), since it will recompute them everytime anything changes on the ui. Usually, render shouldn't re-create more renders. I don't think it will be that much of a performance hog, but maybe try to avoid them in the future (when we add all of them, the perf impact gets bigger)